applications/ConstitutiveLawsApplication/custom_constitutive/associative_plastic_damage_model.cpp

```diff
@@ -81,19 +80,30 @@ void AssociativePlasticDamageModel<TYieldSurfaceType>::CalculateMaterialResponse
         r_constitutive_law_options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
 
         PlasticDamageParameters plastic_damage_parameters = PlasticDamageParameters();
-        InitializePlasticDamageParameters(r_strain_vector, rValues.GetMaterialProperties(),
-            characteristic_length, plastic_damage_parameters);
+        const auto &r_mat_props = rValues.GetMaterialProperties();
+        InitializePlasticDamageParameters(r_strain_vector, r_mat_props, characteristic_length, plastic_damage_parameters);
 
         CheckMinimumFractureEnergy(rValues, plastic_damage_parameters);
 
-        CalculateConstitutiveMatrix(rValues, plastic_damage_parameters);
+        const bool crack_reclosure = (r_mat_props.Has(CRACK_RECLOSING)) ? r_mat_props[CRACK_RECLOSING] : false;
+        if (crack_reclosure) {
+            // here we compute a predictor of the constitutive matrix
+            BoundedVectorType predictor(VoigtSize);
+            CalculateConstitutiveMatrix(rValues, plastic_damage_parameters);
+            noalias(predictor) = prod(plastic_damage_parameters.ConstitutiveMatrix, plastic_damage_parameters.StrainVector - plastic_damage_parameters.PlasticStrain);
+            double tension_parameter, compression_parameter;
+            GenericConstitutiveLawIntegratorPlasticity<TYieldSurfaceType>::CalculateIndicatorsFactors(predictor, tension_parameter,compression_parameter);
+            double det = 0.0;
+            MathUtils<double>::InvertMatrix(tension_parameter*plastic_damage_parameters.ComplianceMatrix+compression_parameter*plastic_damage_parameters.ComplianceMatrixCompression, plastic_damage_parameters.ConstitutiveMatrix, det);
```

Comment:
I don't know the performance impact of doing this operation as it is here. Opinion @RiccardoRossi ?